drivers/soundwire/bus.c

```diff
@@ -2038,3 +2038,37 @@ void sdw_clear_slave_status(struct sdw_bus *bus, u32 request)
 	}
 }
 EXPORT_SYMBOL(sdw_clear_slave_status);
+
+int sdw_bpt_send_async_and_wait(struct sdw_bus *bus, struct sdw_slave *slave,
```

Comment:
iow, sdw_bpt_send_sync() ?

Reply:
After reading the document in the first commit again. I would keep the send async and wait APIs separately.
```
One possible strategy to speed-up all initialization tasks would be to
start a BRA transfer for firmware download, then deal with all the
"regular" read/writes in parallel with the command channel, and last
to wait for the BRA transfers to complete. This would allow for a
degree of overlap instead of a purely sequential solution. As a
results, the BRA API must support async transfers and expose a
separate wait function.
```

Reply:
Sure, but adding a helper to do a sync transfer does not mean that the async and wait cannot be used as described while helping users to avoid duplicated code.
The `do_bpt_sequence()` could be as simple as:
```c
static int do_bpt_sequence(struct sdw_slave *slave, bool write, u8 *buffer)
{
	struct sdw_bpt_msg msg = {0};

	msg.addr = start_addr;
	msg.len = num_bytes;
	msg.dev_num = slave->dev_num;
	if (write)
		msg.flags = SDW_MSG_FLAG_WRITE;
	else
		msg.flags = SDW_MSG_FLAG_READ;
	msg.buf = buffer;

	return sdw_bpt_send_sync(slave->bus, slave, &msg); /* Errors printed in the helper */
}
```